File: ChatBeet/Data/Entities/QuoteMessage.cs
namespace ChatBeet.Data.Entities;

public class QuoteMessage
{
    public required string Content { get; set; }
    public DateTime CreatedAt { get; set; }
    public Guid AuthorId { get; set; }
    public int Embeds { get; set; }
    public int Attachments { get; set; }

    public virtual User? Author { get; set; }
}

File: ChatBeet/ClientApp/src/app/quotes/quote-message.ts
import { User } from "../common/user";
import { UserId } from "../common/user-id";

export class QuoteMessage {
    public content: string;
    public createdAt: Date;
    public authorId: UserId;
    public embeds: number;
    public attachments: number;

    public author: User;

    constructor(data: any) {
        this.content = data.content;
        this.createdAt = new Date(data.createdAt);
        this.authorId = data.authorId;
        this.embeds = data.embeds;
        this.author = new User(data.author);
        this.attachments = data.attachments;
    }

    public attachmentText = (): string => `This message originally had ${this.attachments} ${this.attachments === 1 ? 'attachment' : 'attachments'}.`;
    public embedText = (): string => `This message originally had ${this.embeds} ${this.embeds === 1 ? 'embed' : 'embeds'}.`;
}